CONTRIBUTING.md

```diff
@@ -0,0 +1,254 @@
+  this should be taken care of automatically by the modeline (like the one
+  below).
+
+- No trailing whitespace.
+
+- Use `"` for enclosing strings instead of `'`, unless using `'` avoids
+  escaping of lots of `"`:
+
+  ```javascript
+  alert("foo")
+
+  alert('foo')
+  ```
+
+- Use `//` regexp literals rather than RegExp constructors, unless you're
+  constructing an expression on the fly, or RegExp constructors allow you to
+  escape less `/s` than the additional escaping of special characters required
+  by string quoting:
+
+  ```javascript
+  // Good
+  /application\/xhtml\+xml/
+  RegExp("http://(www\\.)vimperator.org/(.*)/(.*)")
+
+  // Bad
+  RegExp("application/xhtml\\+xml")
+  /http:\/\/(www\.)vimperator.org\/(.*)\/(.*)/
+  ```
+
+- Exactly one space after `if/for/while/catch` etc. and after a comma, but none
+  after a parenthesis or after a function call:
+
+  ```javascript
+  for (pre; condition; post)
+  alert("foo");
+  ```
+
+- Bracing is formatted as follows:
+
+  ```javascript
+  function myFunction () {
+      if (foo)
+          return bar;
+      else {
+          baz = false;
+          return baz;
+      }
+  }
+  var quux = frob("you",
+      {
+          a: 1,
+          b: 42,
+          c: {
+              hoopy: "frood"
+          }
+      });
+  ```
+
+  When in doubt, look for similar code.
+
+- No braces for one-line conditional statements:
+
+  ```javascript
+  if (foo)
+      frob();
+  else
+      unfrob();
+  ```
+
+- Prefer lambda-style functions where suitable:
+
+  ```javascript
+  // Good
+  list.filter(function (elem) elem.good != elem.BAD);
+  list.forEach(function (elem) { window.alert(elem); });
+
+  // Bad
+  list.filter(function (elem) { return elem.good != elem.BAD });
+  list.forEach(function (elem) window.alert(elem));
+  ```
+
+- Anonymous function definitions should be formatted with a space after the
+  keyword `function`:
+
+  ```javascript
+  // Good
+  function () {}
+
+  // Bad
+  function() {}
+  ```
+
+- Prefer the use of `let` over `var` i.e. only use `var` when required.
+
+  For more details, see:
+  https://developer.mozilla.org/en/New_in_JavaScript_1.7#Block_scope_with_let
+
+- Reuse common local variable names. E.g. `elem` is generally used for element,
+  `win` for windows, `func` for functions, `ret` for return values etc.
+
+- Prefer `//` over `/* */` comments (exceptions for big comments are usually
+  OK):
+
+  ```javascript
+  // Good
+  if (HACK) // TODO: remove hack
+
+  // Bad
+  if (HACK) /* TODO: remove hack */
+  ```
+
+- Documentation comment blocks use `/** ... */` Wrap these lines at 80
+  characters.
+
+- Only wrap lines if it makes the code obviously clearer. Lines longer than 132
+  characters should probably be broken up rather than wrapped anyway.
+
+- Use UNIX new lines (`\n`), not windows (`\r\n`) or old Mac ones (`\r`).
+
+- Use `Iterator` or `Array#forEach` to iterate over arrays.
```

Comment:
Sounds good, but could you possibly formulate an updated version? Not really my domain of expertise and I don't want to add any unnecessary ambiguity :\